biojava-structure/src/main/java/org/biojava/nbio/structure/align/ce/CECalculator.java

```diff
@@ -957,9 +955,7 @@ private void checkBestTraces( AFPChain afpChain,
 
 		// removing some loops that are run in orig CE
 		// and which did not do anything
-		if ( debug ){
-			checkPrintRmsdNew(traceMaxSize, winSize, ca1, ca2);
-		}
+		checkPrintRmsdNew(traceMaxSize, winSize, ca1, ca2);
```

Comment:
> On 20 May 2016, at 03:05, Jose Manuel Duarte <notifications@github.com> wrote:
> 
> In biojava-structure/src/main/java/org/biojava/nbio/structure/align/ce/CECalculator.java:
> 
> > @@ -957,9 +955,7 @@ private void checkBestTraces( AFPChain afpChain,
> >  
> >  		// removing some loops that are run in orig CE
> >  		// and which did not do anything
> > -		if ( debug ){
> > -			checkPrintRmsdNew(traceMaxSize, winSize, ca1, ca2);
> > -		}
> > +		checkPrintRmsdNew(traceMaxSize, winSize, ca1, ca2);
> 
> Here the refactoring is altering what the code actually does: now the checkPrintRmsdNew method will run always (and it performs some calculations that will make things slower). The only way that I see to avoid it is using isDebugEnabled.
> 

If this is Debug-only code then it shouldn’t still be in here. If it’s testing functionality then it should be in a test. If it’s logging then it should just be logging statements. If it’s doing something else then it shouldn’t be.




-- 
The University of Edinburgh is a charitable body, registered in
Scotland, with registration number SC005336.